### ISSUE TITLE
Getting NAN value for p1 and p2

### ISSUE THREAD

**Issue by anilkoeri** [markrogoyski/math-php#393]
I am getting NAN value for p1 and P2 for the below function. Could you please let me know the issue? 
// T test - Two samples (from sample data)
$x₁    = [27.5, 21.0, 19.0, 23.6, 17.0, 17.9, 16.9, 20.1, 21.9, 22.6, 23.1, 19.6, 19.0, 21.7, 21.4];
$x₂    = [27.1, 22.0, 20.8, 23.4, 23.4, 23.5, 25.8, 22.0, 24.8, 20.2, 21.9, 22.1, 22.9, 20.5, 24.4];
$tTest = Significance::tTest($x₁, $x₂);
print_r($tTest);

I have attached the used array data in text file.
[data.txt](https://github.com/markrogoyski/math-php/files/5344189/data.txt)

**Comment by Beakerboy** [comment]
Have you had success using the same data In another piece of software? What were the results?

**Comment by markrogoyski** [comment]
This is the result from R
```R
> t.test(x1, x2)

	Welch Two Sample t-test

data:  x1 and x2
t = 0.18973, df = 548.82, p-value = 0.8496
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -8.187625  9.938437
sample estimates:
mean of x mean of y 
 50.29022  49.41481 
```

This is MathPHP
```php
(
    [t] => 0.18973304760123
    [df] => 548.8156531233
    [p1] => NAN
    [p2] => NAN
    [mean1] => 50.290220820189
    [mean2] => 49.414814814815
    [sd1] => 52.741226989123
    [sd2] => 58.125026785483
)
```

The summary data and t and df are right. To determine p it is using the Student's T distribution. I think the Student T distribution is using the regularizedIncompleteBeta function, and that is where the NAN is coming from.

**Comment by anilkoeri** [comment]
@Beakerboy  Yes, Tried with Microsoft excel and it is 0.15 for P1.  then also used zTestTwoSample and got the almost same result as Microsoft excel. 

@markrogoyski  I am getting the same result as yours

Array
(
    [t] => 0.18973304760123
    [df] => 548.8156531233
    [p1] => NAN
    [p2] => NAN
    [mean1] => 50.290220820189
    [mean2] => 49.414814814815
    [sd1] => 52.741226989123
    [sd2] => 58.125026785483
)

**Comment by markrogoyski** [comment]
I think the root problem is in the regularized incomplete beta function, which is what the StudentT distribution's CDF is using.

Using the input data provided by @anilkoeri, this is the input going into the incomplete beta function, and this unit test reproduces the issue.
```php
    /**
     * @test Github Issue 393 Bug
     * Reference expected values:
     *   Python scipy.special.betainc(a, b, x)
     *     >>> scipy.special.betainc(274.40782656165, 0.5, 0.99993441100298)
     *     0.8495884744315958
     *   Online calculator https://keisan.casio.com/exec/system/1180573396
     *     0.8495884744316587246283
     */
    public function testIssue393BugInRegularizedIncompleteBeta()
    {
        // Given
        $x = 0.99993441100298;
        $a = 274.40782656165;
        $b = 0.5;

        // And
        $expected = 0.8495884744315958;

        // When
        $betainc = Special::regularizedIncompleteBeta($x, $a, $b);

        // Then
        $this->assertEquals($expected, $betainc, '', '0.00001');
    }
```

The actual result is a NAN.
```
Failed asserting that NAN matches expected 0.8495884744315958.
```

SciPy and a high-precision online calculator both agree that the result should be approximately 0.8495884744315958.

@Beakerboy,
Do you mind taking a look at this? I think you implemented the incomplete beta function and know more about it than I do.

I added some more unit test cases for it today and it seems to work for most inputs. However, it seems there might be some edge cases where it is getting NAN. You can use the unit test I provided above to reproduce the issue.

Also, here is another unit test that exhibits a different issue--this time, division by zero. This is a unit test from SciPy.
```php
    /**
     * @test SciPy unit test: https://github.com/scipy/scipy/blob/da50c1c435f3588e9d8d8cac22a1383557ed6046/scipy/special/tests/test_basic.py#L170
     * Reference expected values:
     *   Python scipy.special.betainc(a, b, x)
     *     >>> scipy.special.betainc(0.0342, 171, 1e-10)
     *     0.5526991690180665
     *   Online calculator https://keisan.casio.com/exec/system/1180573396
     *     0.5526991690180709217802
     */
    public function testSciPyUnitTestBetaIncAllClose()
    {
        // Given
        $x = 1e-10;
        $a = 0.0342;
        $b = 171;

        // And
        $expected = 0.5526991690180665;

        // When
        $betainc = Special::regularizedIncompleteBeta($x, $a, $b);

        // Then
        $this->assertEquals($expected, $betainc, '', '0.00001');
    }
```

```
There was 1 error:

1) MathPHP\Tests\Functions\SpecialTest::testSciPyUnitTestBetaIncAllClose
Division by zero

/src/Functions/Special.php:640
/src/Functions/Special.php:725
/src/Functions/Special.php:737
```

Thanks,
Mark

**Comment by markrogoyski** [comment]
Hi @anilkoeri,

This has been fixed in [Version 2.5.0](https://github.com/markrogoyski/math-php/releases/tag/v2.5.0).

Thank you for reporting this bug. And thank you for your interest in MathPHP.

Mark
